### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by ClintonYounge**
# Update
- Install Rspec.
- Implemented Unit Testings for all classes.

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  it 'should have a title' do
+    expect(@book.title).to eq('Title')
+  end
+
+  it 'should have an author' do
+    expect(@book.author).to eq('Author')
+  end
+end
```
Comment (laiifuu) [inline spec/book_spec.rb]: As mentionnd in the requirements, kindly add unit tests for the methods that exist in the `Book` class: 
- add_rental

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  it 'should have a name' do
+    expect(@person.name).to eq('Alice')
+  end
+
+  it 'should have an age' do
+    expect(@person.age).to eq(15)
+  end
+end
```
Comment (laiifuu) [inline spec/person_spec.rb]: As mentioned in the requirements, you should add unit tests for methods that exist in the Person class: 
- generate_new_id
- add_rental
- correct_name
- can_use_services?

File: spec/teacher_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  it 'validate the teacher age' do
+    expect(@teacher.age).to eq(18)
+  end
+
+  it 'validate empty list of rentals' do
+    expect(@teacher.rentals).to eq([])
+  end
+end
```
Comment (laiifuu) [inline spec/teacher_spec.rb]: As mentioned in the requirements, you should add unit tests for the methods that exist in the `Teacher` class: 
- can_use_services?

File: spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  it 'validate the student age' do
+    expect(@student.age).to eq(18)
+  end
+
+  it 'validate empty list of rentals' do
+    expect(@student.rentals).to eq([])
+  end
+end
```
Comment (laiifuu) [inline spec/student_spec.rb]: As mentioned in the requirements, you should add unit tests for the methods that exist in the `Student` class:
- assign_classroom
- play_hookey

**Review by laiifuu — CHANGES_REQUESTED** [review]
## Hi team :wave: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
- [x] Rspec gem added to the Gemfile :+1: 
- [x] Green linters check :heavy_check_mark: 

## Required Changes ♻️
- [ ] As mentionned in the requirements, kindly add unit tests for the `TrimmerDecorator` class and the `CapitalizeDecorator` class. 
_Check the comments under the review. :point_down:_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @ClintonYounge, @MoyasiGinko  & @sylvester-wamaya 

Great work on making the changes requested by a previous reviewer 👏! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ All implemented tests are passing
- ✅ `rspec` gem is used
- ✅ Correct Linters

## Required Changes 🛑

- [ ] Well done for implementing almost all suggested tests 👏 However, curently, you are missing the tests for the `TrimmerDecorator` and the `CapitalizeDecorator` class as suggested by the previous reviewer in the PR summary 📌✨Kindly have a look 👇and please make sure to implement these tests

image


_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by peterrobert — APPROVED** [review]
Hi @ClintonYounge, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**